### DISCUSSION TITLE
Add weights to tests to compute weighted sum of passed & failed tests

### DISCUSSION THREAD

**Discussion by ycopin** [pytest-dev/pytest#12906] [Q&A] [answered]
Hi, I'm writing a small pyTestExam plugin (not yet structured as it) to test and grade computer exams, see https://gitlab.in2p3.fr/ycopin/pyTestExam. I can count the passed and failed tests (see `pytest_terminal_summary` in https://gitlab.in2p3.fr/ycopin/pyTestExam/-/blob/master/conftest.py), but I would now like to add *weights* to the different tests, as they are not all of equal difficulty. 
I guess I would use the mark mechanism for the weighting:
```
@pytest.mark.weight(2)
```
but I did not find how to access the individual test weights in the final  `pytest_terminal_summary` to compute a *weighted sum* of passed and failed tests. Help and comments are welcome!

**Comment by RonnyPfannschmidt** [comment]
this is outside of the scope of pytest core itself

however it would be a perfect fit for a custom plugin that would output grading stats and details 

the test weights are typically no longer available at the final stat step, they need to be transferred when the report is handled

this however practically requires to redo the stats the terminal plugin did

**Reply by ycopin** [reply]
Thanks for the feedback. I understand this is outside the scope of pytest, hence the plugin format. 

Any guidance on the way to properly "propagate" the weights up to the final report, and make the final weighted sum?

> this however practically requires to redo the stats the terminal plugin did

Sorry, I don't get it: are you referring to a specific plugin that does something similar, and which I could use as a guidance?

**Reply by RonnyPfannschmidt** [reply]
The built-in terminal report plugin does stats, you'd have to do something similar that's weight aware

**Reply by ycopin** [reply]
mmmm.... Thank you for your answer, but that does not really help me in practice. Do you have more specific guidance?

**Comment by ycopin** [comment, marked as answer]
For the record, this is now implemented in https://gitlab.in2p3.fr/ycopin/pyTestExam/ v1.0 through markers.
